src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java

```diff
@@ -478,23 +473,38 @@ private Slave prepareDockerPlatforms() throws Exception {
     }
 
     @Test public void binaryCaching() throws Exception {
-        assumeTrue(!Objects.equals(System.getenv().get("SKIP_DURABLE_TASK_BINARY_GENERATION"), "true") && !platform.equals(TestPlatform.UBUNTU_NO_BINARY));
+        assumeFalse(platform.equals(TestPlatform.UBUNTU_NO_BINARY));
         String os;
+        String architecture;
+        String arch = System.getProperty("os.arch");
```

Comment:
nit: It's kind of confusing having variables named both `arch` and `architecture`. IIUC, `architecture` is used for binary selection and `arch` is just used to figure out if we are on an ARM-based Mac. If that's right, I might move `String arch = System.getProperty("os.arch");` down right after `os = "darwin";` below so it's clearer that it is only used for the ARM check.

Reply:
hopefully un-confused in fcab5fac0547fd470de5427ad1b6832cc3154614